src/containers/Signup/Signup.tsx

```diff
@@ -0,0 +1,51 @@
+import gql from 'graphql-tag';
+import jwtDecode from 'jwt-decode';
+import React, { useState } from 'react';
+import { useApolloClient, useMutation } from '@apollo/react-hooks';
+import AuthForm from '../../components/AuthForm/AuthForm';
+
+const SIGN_UP = gql`
+  mutation signup($name: String!, $email: String!, $password: String!) {
+    signup(name: $name, email: $email, password: $password)
+  }
+`;
+
+const Signup: React.FC = () => {
+  const client = useApolloClient();
+  const [inputs, setInputs] = useState({ name: '', email: '', password: '' });
+  const [errorMsg, setErrorMsg] = useState('');
+  const [signup] = useMutation(SIGN_UP, {
+    variables: { name: inputs.name, email: inputs.email, password: inputs.password },
+  });
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    // TODO: any??
+    const res: any = await signup();
+    if (res.data.signup) {
+      localStorage.setItem('token', res.data.signup);
```

Comment:
All this logic related to the signup which receives a token, set it into the local storage etc, should be completely extracted in an external service to reuse the service across multiple pages and avoid code duplication. Also, having this logic here contaminate your component with additional logic that shouldn't be handled here. Consider to refactoring the authentication to provide signup/signin/signout method from a centralized service.